src/pypowsybl_jupyter/networkexplorer.py

```diff
@@ -126,32 +138,60 @@ def on_text_changed(d):
         value=sel_ctx.get_selected(),
         description='Found',
         disabled=False,
-        layout=widgets.Layout(width='350px', height='670px')
+        layout=widgets.Layout(flex='80%', height='100%', width='350px', margin='0 0 0 0')
     )
 
     def on_selected(d):
         if d['new'] != None:
             sel_ctx.set_selected(d['new'])
-            update_nad_diagram()
-            update_sld_diagram()
+            update_explorer()
 
     found.observe(on_selected, names='value')
 
-    update_nad_diagram()
-    update_sld_diagram()
+    history = widgets.Select(
+        options=sel_ctx.get_history_as_list(),
+        value=sel_ctx.get_selected(),
+        description='History',
+        disabled=False,
+        layout=widgets.Layout(flex='18%', height='100%', width='350px', margin='1px 0 0 0')
+    )
+
+    def on_selected_history(d):
+        if d['new'] != None:
+            sel_ctx.set_selected(d['new'])
+            update_explorer()
+
+    history.observe(on_selected_history, names='value')
+
+    def update_explorer():
+        sel=sel_ctx.get_selected()
+
+        update_select_widget(found, sel if sel_ctx.is_selected_in_filtered_vls() else None, None, on_selected)
+        update_select_widget(history, sel, sel_ctx.get_history_as_list(), on_selected_history)
```

Comment:
If we arrive here from the history, we should not update the history, it should stay as it is